py develop-inst-noop: /home/newton/work/orion
py installed: appdirs==1.4.4,asttokens==2.0.8,async-generator==1.10,attrs==22.1.0,certifi==2022.9.24,charset-normalizer==2.1.1,cloudpickle==2.2.0,codefind==0.1.3,coverage==6.5.0,exceptiongroup==1.0.0rc9,execnet==1.9.0,executing==0.8.3,falcon==3.1.0,falcon-cors==1.1.7,filelock==3.8.0,gitdb==4.0.9,GitPython==3.1.29,giving==0.4.0,gunicorn==20.1.0,h11==0.14.0,idna==3.4,iniconfig==1.1.1,joblib==1.2.0,kaleido==0.2.1,numpy==1.23.3,-e git+ssh://git@github.com/Delaunay/orion.git@9d51d91b177fb935ae7d7f11a99adb6135e7474b#egg=orion,orion.algo.gradient-descent @ file:///home/newton/work/orion/tests/functional/gradient_descent_algo,outcome==1.2.0,packaging==21.3,pandas==1.4.4,plotly==5.10.0,pluggy==1.0.0,psutil==5.9.2,ptera==1.4.0,pure-eval==0.2.2,py==1.11.0,pyaml==21.10.1,pymongo==4.2.0,pyparsing==3.0.9,PySocks==1.7.1,pytest==7.1.3,pytest-custom-exit-code==0.3.0,pytest-forked==1.4.0,pytest-lazy-fixture==0.6.3,pytest-mock==3.10.0,pytest-timeout==2.1.0,pytest-xdist==2.5.0,python-dateutil==2.8.2,pytz==2022.4,PyYAML==6.0,reactivex==4.0.4,requests==2.28.1,scikit-learn==1.1.2,scikit-optimize==0.9.0,scipy==1.9.2,selenium==4.5.0,six==1.16.0,smmap==5.0.0,sniffio==1.3.0,sortedcontainers==2.4.0,tabulate==0.9.0,tenacity==8.1.0,threadpoolctl==3.1.0,tomli==2.0.1,trio==0.22.0,trio-websocket==0.9.2,typing_extensions==4.4.0,urllib3==1.26.12,varname==0.8.3,wsproto==1.2.0
py run-test-pre: PYTHONHASHSEED='2501674067'
py run-test: commands[0] | pip install -U /home/newton/work/orion/tests/functional/gradient_descent_algo
Processing ./tests/functional/gradient_descent_algo
  Preparing metadata (setup.py): started
  Preparing metadata (setup.py): finished with status 'done'
Requirement already satisfied: orion in ./src (from orion.algo.gradient-descent==0.1) (0.1.2.post3112+g790e4632)
Requirement already satisfied: cloudpickle in ./.tox/py/lib/python3.9/site-packages (from orion->orion.algo.gradient-descent==0.1) (2.2.0)
Requirement already satisfied: PyYAML in ./.tox/py/lib/python3.9/site-packages (from orion->orion.algo.gradient-descent==0.1) (6.0)
Requirement already satisfied: pymongo>=3 in ./.tox/py/lib/python3.9/site-packages (from orion->orion.algo.gradient-descent==0.1) (4.2.0)
Requirement already satisfied: numpy in ./.tox/py/lib/python3.9/site-packages (from orion->orion.algo.gradient-descent==0.1) (1.23.3)
Requirement already satisfied: scipy in ./.tox/py/lib/python3.9/site-packages (from orion->orion.algo.gradient-descent==0.1) (1.9.2)
Requirement already satisfied: gitpython in ./.tox/py/lib/python3.9/site-packages (from orion->orion.algo.gradient-descent==0.1) (3.1.29)
Requirement already satisfied: filelock in ./.tox/py/lib/python3.9/site-packages (from orion->orion.algo.gradient-descent==0.1) (3.8.0)
Requirement already satisfied: tabulate in ./.tox/py/lib/python3.9/site-packages (from orion->orion.algo.gradient-descent==0.1) (0.9.0)
Requirement already satisfied: AppDirs in ./.tox/py/lib/python3.9/site-packages (from orion->orion.algo.gradient-descent==0.1) (1.4.4)
Requirement already satisfied: plotly in ./.tox/py/lib/python3.9/site-packages (from orion->orion.algo.gradient-descent==0.1) (5.10.0)
Requirement already satisfied: kaleido in ./.tox/py/lib/python3.9/site-packages (from orion->orion.algo.gradient-descent==0.1) (0.2.1)
Requirement already satisfied: requests in ./.tox/py/lib/python3.9/site-packages (from orion->orion.algo.gradient-descent==0.1) (2.28.1)
Requirement already satisfied: pandas<=1.4.4 in ./.tox/py/lib/python3.9/site-packages (from orion->orion.algo.gradient-descent==0.1) (1.4.4)
Requirement already satisfied: gunicorn in ./.tox/py/lib/python3.9/site-packages (from orion->orion.algo.gradient-descent==0.1) (20.1.0)
Requirement already satisfied: falcon in ./.tox/py/lib/python3.9/site-packages (from orion->orion.algo.gradient-descent==0.1) (3.1.0)
Requirement already satisfied: falcon-cors in ./.tox/py/lib/python3.9/site-packages (from orion->orion.algo.gradient-descent==0.1) (1.1.7)
Requirement already satisfied: scikit-learn in ./.tox/py/lib/python3.9/site-packages (from orion->orion.algo.gradient-descent==0.1) (1.1.2)
Requirement already satisfied: psutil in ./.tox/py/lib/python3.9/site-packages (from orion->orion.algo.gradient-descent==0.1) (5.9.2)
Requirement already satisfied: joblib in ./.tox/py/lib/python3.9/site-packages (from orion->orion.algo.gradient-descent==0.1) (1.2.0)
Requirement already satisfied: pytest>=3.0.0 in ./.tox/py/lib/python3.9/site-packages (from orion->orion.algo.gradient-descent==0.1) (7.1.3)
Requirement already satisfied: scikit-optimize in ./.tox/py/lib/python3.9/site-packages (from orion->orion.algo.gradient-descent==0.1) (0.9.0)
Requirement already satisfied: pytz>=2020.1 in ./.tox/py/lib/python3.9/site-packages (from pandas<=1.4.4->orion->orion.algo.gradient-descent==0.1) (2022.4)
Requirement already satisfied: python-dateutil>=2.8.1 in ./.tox/py/lib/python3.9/site-packages (from pandas<=1.4.4->orion->orion.algo.gradient-descent==0.1) (2.8.2)
Requirement already satisfied: tomli>=1.0.0 in ./.tox/py/lib/python3.9/site-packages (from pytest>=3.0.0->orion->orion.algo.gradient-descent==0.1) (2.0.1)
Requirement already satisfied: attrs>=19.2.0 in ./.tox/py/lib/python3.9/site-packages (from pytest>=3.0.0->orion->orion.algo.gradient-descent==0.1) (22.1.0)
Requirement already satisfied: iniconfig in ./.tox/py/lib/python3.9/site-packages (from pytest>=3.0.0->orion->orion.algo.gradient-descent==0.1) (1.1.1)
Requirement already satisfied: packaging in ./.tox/py/lib/python3.9/site-packages (from pytest>=3.0.0->orion->orion.algo.gradient-descent==0.1) (21.3)
Requirement already satisfied: pluggy<2.0,>=0.12 in ./.tox/py/lib/python3.9/site-packages (from pytest>=3.0.0->orion->orion.algo.gradient-descent==0.1) (1.0.0)
Requirement already satisfied: py>=1.8.2 in ./.tox/py/lib/python3.9/site-packages (from pytest>=3.0.0->orion->orion.algo.gradient-descent==0.1) (1.11.0)
Requirement already satisfied: gitdb<5,>=4.0.1 in ./.tox/py/lib/python3.9/site-packages (from gitpython->orion->orion.algo.gradient-descent==0.1) (4.0.9)
Requirement already satisfied: setuptools>=3.0 in ./.tox/py/lib/python3.9/site-packages (from gunicorn->orion->orion.algo.gradient-descent==0.1) (65.3.0)
Requirement already satisfied: tenacity>=6.2.0 in ./.tox/py/lib/python3.9/site-packages (from plotly->orion->orion.algo.gradient-descent==0.1) (8.1.0)
Requirement already satisfied: urllib3<1.27,>=1.21.1 in ./.tox/py/lib/python3.9/site-packages (from requests->orion->orion.algo.gradient-descent==0.1) (1.26.12)
Requirement already satisfied: certifi>=2017.4.17 in ./.tox/py/lib/python3.9/site-packages (from requests->orion->orion.algo.gradient-descent==0.1) (2022.9.24)
Requirement already satisfied: idna<4,>=2.5 in ./.tox/py/lib/python3.9/site-packages (from requests->orion->orion.algo.gradient-descent==0.1) (3.4)
Requirement already satisfied: charset-normalizer<3,>=2 in ./.tox/py/lib/python3.9/site-packages (from requests->orion->orion.algo.gradient-descent==0.1) (2.1.1)
Requirement already satisfied: threadpoolctl>=2.0.0 in ./.tox/py/lib/python3.9/site-packages (from scikit-learn->orion->orion.algo.gradient-descent==0.1) (3.1.0)
Requirement already satisfied: pyaml>=16.9 in ./.tox/py/lib/python3.9/site-packages (from scikit-optimize->orion->orion.algo.gradient-descent==0.1) (21.10.1)
Requirement already satisfied: smmap<6,>=3.0.1 in ./.tox/py/lib/python3.9/site-packages (from gitdb<5,>=4.0.1->gitpython->orion->orion.algo.gradient-descent==0.1) (5.0.0)
Requirement already satisfied: six>=1.5 in ./.tox/py/lib/python3.9/site-packages (from python-dateutil>=2.8.1->pandas<=1.4.4->orion->orion.algo.gradient-descent==0.1) (1.16.0)
Requirement already satisfied: pyparsing!=3.0.5,>=2.0.2 in ./.tox/py/lib/python3.9/site-packages (from packaging->pytest>=3.0.0->orion->orion.algo.gradient-descent==0.1) (3.0.9)
Building wheels for collected packages: orion.algo.gradient-descent
  Building wheel for orion.algo.gradient-descent (setup.py): started
  Building wheel for orion.algo.gradient-descent (setup.py): finished with status 'done'
  Created wheel for orion.algo.gradient-descent: filename=orion.algo.gradient_descent-0.1-py2.py3-none-any.whl size=3802 sha256=d30123cd282401a133a4b693c336589fcfbd82ee913db25be0a730fadbc3d971
  Stored in directory: /tmp/pip-ephem-wheel-cache-lcx91c_m/wheels/20/1f/5e/ee04c2e8c8df32b66c17fa8ff00b179a7e9a91a490336d4134
Successfully built orion.algo.gradient-descent
Installing collected packages: orion.algo.gradient-descent
  Attempting uninstall: orion.algo.gradient-descent
    Found existing installation: orion.algo.gradient-descent 0.1
    Uninstalling orion.algo.gradient-descent-0.1:
      Successfully uninstalled orion.algo.gradient-descent-0.1
Successfully installed orion.algo.gradient-descent-0.1

[notice] A new release of pip available: 22.2.2 -> 22.3
[notice] To update, run: pip install --upgrade pip
py run-test: commands[1] | coverage run --source=src --parallel-mode -m pytest --durations=50 --durations-min 1 -vv --timeout=360 --ignore tests/unittests/algo/long --ignore tests/functional/backward_compatibility --ignore tests/functional/serving/test_frontend.py --ignore tests/stress --lf --color=no -x -s --log-cli-level=Debug
============================= test session starts ==============================
platform linux -- Python 3.9.5, pytest-7.1.3, pluggy-1.0.0 -- /home/newton/work/orion/.tox/py/bin/python
cachedir: .tox/py/.pytest_cache
rootdir: /home/newton/work/orion, configfile: tox.ini
plugins: mock-3.10.0, timeout-2.1.0, forked-1.4.0, lazy-fixture-0.6.3, xdist-2.5.0, custom-exit-code-0.3.0
timeout: 360.0s
timeout method: signal
timeout func_only: False
collecting ... collected 26 items / 23 deselected / 3 selected
run-last-failure: rerun previous 3 failures (skipped 112 files)

tests/unittests/client/test_runner.py::test_runner_inside_thread 
-------------------------------- live log call ---------------------------------
DEBUG    orion.core.utils:__init__.py:78 Found a dask BaseExecutor from distribution: orion=0.1.2.post3112+g790e4632
DEBUG    orion.core.utils:__init__.py:78 Found a joblib BaseExecutor from distribution: orion=0.1.2.post3112+g790e4632
DEBUG    orion.core.utils:__init__.py:78 Found a singleexecutor BaseExecutor from distribution: orion=0.1.2.post3112+g790e4632
WARNING  orion.client.runner:runner.py:64 SIGINT/SIGTERM protection hooks could not be installed because Runner is executing inside a thread/subprocess, results could get lost on interruptions
DEBUG    orion.client.runner:runner.py:361 Sampled 2 new configs
DEBUG    orion.client.runner:runner.py:388 Scheduled new trials
DEBUG    orion.client.runner:runner.py:399 Gathered new results 2
WARNING  orion.client.runner:runner.py:64 SIGINT/SIGTERM protection hooks could not be installed because Runner is executing inside a thread/subprocess, results could get lost on interruptions
DEBUG    orion.client.runner:runner.py:361 Sampled 2 new configs
DEBUG    orion.client.runner:runner.py:388 Scheduled new trials
DEBUG    orion.client.runner:runner.py:399 Gathered new results 1
WARNING  orion.client.runner:runner.py:64 SIGINT/SIGTERM protection hooks could not be installed because Runner is executing inside a thread/subprocess, results could get lost on interruptions
DEBUG    orion.client.runner:runner.py:361 Sampled 1 new configs
DEBUG    orion.client.runner:runner.py:388 Scheduled new trials
DEBUG    orion.client.runner:runner.py:399 Gathered new results 1
WARNING  orion.client.runner:runner.py:64 SIGINT/SIGTERM protection hooks could not be installed because Runner is executing inside a thread/subprocess, results could get lost on interruptions
DEBUG    orion.client.runner:runner.py:361 Sampled 1 new configs
DEBUG    orion.client.runner:runner.py:388 Scheduled new trials
DEBUG    orion.client.runner:runner.py:399 Gathered new results 1
WARNING  orion.client.runner:runner.py:64 SIGINT/SIGTERM protection hooks could not be installed because Runner is executing inside a thread/subprocess, results could get lost on interruptions
DEBUG    orion.client.runner:runner.py:361 Sampled 1 new configs
DEBUG    orion.client.runner:runner.py:388 Scheduled new trials
WARNING  orion.client.runner:runner.py:64 SIGINT/SIGTERM protection hooks could not be installed because Runner is executing inside a thread/subprocess, results could get lost on interruptions
DEBUG    orion.client.runner:runner.py:399 Gathered new results 1
WARNING  orion.client.runner:runner.py:64 SIGINT/SIGTERM protection hooks could not be installed because Runner is executing inside a thread/subprocess, results could get lost on interruptions
DEBUG    orion.client.runner:runner.py:361 Sampled 1 new configs
DEBUG    orion.client.runner:runner.py:388 Scheduled new trials
DEBUG    orion.client.runner:runner.py:399 Gathered new results 1
Traceback (most recent call last):
  File "/home/newton/work/orion/tests/unittests/client/test_runner.py", line 595, in run_runner
    runner.run()
  File "/home/newton/work/orion/src/orion/client/runner.py", line 306, in run
    gathered = self.gather()
  File "/home/newton/work/orion/src/orion/client/runner.py", line 404, in gather
    trial = self.pending_trials.pop(result.future)
KeyError: <orion.executor.multiprocess_backend._Future object at 0x7fcc521cc4c0>
FAILED

=================================== FAILURES ===================================
__________________________ test_runner_inside_thread ___________________________

    def test_runner_inside_thread():
        """Runner can execute inside a thread"""
    
        class GetResult:
            def __init__(self) -> None:
                self.r = None
    
            def run(self):
                self.r = run_runner()
    
        result = GetResult()
        thread = Thread(target=result.run)
        thread.start()
        thread.join()
    
>       assert result.r == 0
E       assert 1 == 0
E        +  where 1 = <test_runner.test_runner_inside_thread.<locals>.GetResult object at 0x7fcc52576100>.r

tests/unittests/client/test_runner.py:671: AssertionError
------------------------------ Captured log call -------------------------------
DEBUG    orion.core.utils:__init__.py:78 Found a dask BaseExecutor from distribution: orion=0.1.2.post3112+g790e4632
DEBUG    orion.core.utils:__init__.py:78 Found a joblib BaseExecutor from distribution: orion=0.1.2.post3112+g790e4632
DEBUG    orion.core.utils:__init__.py:78 Found a singleexecutor BaseExecutor from distribution: orion=0.1.2.post3112+g790e4632
WARNING  orion.client.runner:runner.py:64 SIGINT/SIGTERM protection hooks could not be installed because Runner is executing inside a thread/subprocess, results could get lost on interruptions
DEBUG    orion.client.runner:runner.py:361 Sampled 2 new configs
DEBUG    orion.client.runner:runner.py:388 Scheduled new trials
DEBUG    orion.client.runner:runner.py:399 Gathered new results 2
WARNING  orion.client.runner:runner.py:64 SIGINT/SIGTERM protection hooks could not be installed because Runner is executing inside a thread/subprocess, results could get lost on interruptions
DEBUG    orion.client.runner:runner.py:361 Sampled 2 new configs
DEBUG    orion.client.runner:runner.py:388 Scheduled new trials
DEBUG    orion.client.runner:runner.py:399 Gathered new results 1
WARNING  orion.client.runner:runner.py:64 SIGINT/SIGTERM protection hooks could not be installed because Runner is executing inside a thread/subprocess, results could get lost on interruptions
DEBUG    orion.client.runner:runner.py:361 Sampled 1 new configs
DEBUG    orion.client.runner:runner.py:388 Scheduled new trials
DEBUG    orion.client.runner:runner.py:399 Gathered new results 1
WARNING  orion.client.runner:runner.py:64 SIGINT/SIGTERM protection hooks could not be installed because Runner is executing inside a thread/subprocess, results could get lost on interruptions
DEBUG    orion.client.runner:runner.py:361 Sampled 1 new configs
DEBUG    orion.client.runner:runner.py:388 Scheduled new trials
DEBUG    orion.client.runner:runner.py:399 Gathered new results 1
WARNING  orion.client.runner:runner.py:64 SIGINT/SIGTERM protection hooks could not be installed because Runner is executing inside a thread/subprocess, results could get lost on interruptions
DEBUG    orion.client.runner:runner.py:361 Sampled 1 new configs
DEBUG    orion.client.runner:runner.py:388 Scheduled new trials
WARNING  orion.client.runner:runner.py:64 SIGINT/SIGTERM protection hooks could not be installed because Runner is executing inside a thread/subprocess, results could get lost on interruptions
DEBUG    orion.client.runner:runner.py:399 Gathered new results 1
WARNING  orion.client.runner:runner.py:64 SIGINT/SIGTERM protection hooks could not be installed because Runner is executing inside a thread/subprocess, results could get lost on interruptions
DEBUG    orion.client.runner:runner.py:361 Sampled 1 new configs
DEBUG    orion.client.runner:runner.py:388 Scheduled new trials
DEBUG    orion.client.runner:runner.py:399 Gathered new results 1
=============================== warnings summary ===============================
src/orion/benchmark/task/profet/profet_task.py:25
  /home/newton/work/orion/src/orion/benchmark/task/profet/profet_task.py:25: RuntimeWarning: The `profet` extras needs to be installed in order to use the Profet tasks.
  Error: No module named 'torch'
  Use `pip install orion[profet]` to install the profet extras.
    warnings.warn(

src/orion/benchmark/task/profet/model_utils.py:32
  /home/newton/work/orion/src/orion/benchmark/task/profet/model_utils.py:32: RuntimeWarning: The `profet` extras needs to be installed in order to use the Profet tasks.
  Error: No module named 'GPy'
  Use `pip install orion[profet]` to install the profet extras.
    warnings.warn(RuntimeWarning(_ERROR_MSG.format(err)))

tests/unittests/client/test_runner.py::test_runner_inside_thread
  /home/newton/work/orion/src/orion/executor/joblib_backend.py:8: DeprecationWarning: Joblib is deprecated and will be removed in v0.4.0.Use PoolExecutor instead.
    warnings.warn(

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
============================= slowest 50 durations =============================

(3 durations < 1s hidden.  Use -vv to show these durations.)
=========================== short test summary info ============================
FAILED tests/unittests/client/test_runner.py::test_runner_inside_thread - ass...
!!!!!!!!!!!!!!!!!!!!!!!!!! stopping after 1 failures !!!!!!!!!!!!!!!!!!!!!!!!!!!
================= 1 failed, 23 deselected, 3 warnings in 1.45s =================
ERROR: InvocationError for command /home/newton/work/orion/.tox/py/bin/coverage run --source=src --parallel-mode -m pytest --durations=50 --durations-min 1 -vv --timeout=360 --ignore tests/unittests/algo/long --ignore tests/functional/backward_compatibility --ignore tests/functional/serving/test_frontend.py --ignore tests/stress --lf --color=no -x -s --log-cli-level=Debug (exited with code 1)
___________________________________ summary ____________________________________
ERROR:   py: commands failed
